client/src/featureform/serving.py

```diff
@@ -207,11 +228,15 @@ def add_feature_dfs_to_list(self, feature_variant_list, entity_id):
         for feature_variant in feature_variant_list:
             feature = self.db.get_feature_variant(feature_variant[0], feature_variant[1])
             source_name, source_variant = feature['source_name'], feature['source_variant']
-            if self.db.is_transformation(source_name, source_variant):
+            if self.db.is_transformation(source_name, source_variant) != "PRIMARY":
                 feature_df = self.process_transformation(source_name, source_variant)
                 if isinstance(feature_df, pd.Series):
                     feature_df = feature_df.to_frame()
                     feature_df.reset_index(inplace=True)
+                if not entity_id in feature_df.columns:
+                    raise ValueError(f"No column named {entity_id} exists")
```

Comment:
Change to "Could not set entity column. No column name {entity_id} exists in {source_name}-{source_variant}"